src/Marketing.ViewModelComposition/OrdersListGetHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace Marketing.ViewModelComposition
+{
+    using System.Linq;
+    using System.Net.Http;
+    using System.Collections.Generic;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Routing;
+    using ITOps.ViewModelComposition.Json;
+    using ITOps.ViewModelComposition;
+    using EShop.Messages.ViewModelCompositionEvents;
+
+    class OrdersListGetHandler : ISubscribeToViewModelCompositionEvent
+    {
+        public bool Matches(RouteData routeData, string httpVerb, HttpRequest request)
+        {
+            var controller = (string)routeData.Values["controller"];
+            var action = (string)routeData.Values["action"];
+
+            return HttpMethods.IsGet(httpVerb)
+                   && controller.ToLowerInvariant() == "orders"
+                   && action.ToLowerInvariant() == "index";
+        }
+        
+        public void RegisterCallback(DynamicViewModel viewModel)
+        {
+            viewModel.RegisterCallback<OrdersLoaded>(async (pageViewModel, eventData, routeData, query) =>
+            {
+                var orderIds = string.Join(",", eventData.OrdersDictionary.Keys.ToArray());
+                var url = $"http://localhost:50688/product/order?orderIds={orderIds}";
+                var client = new HttpClient();
+                var response = await client.GetAsync(url);
+                dynamic productList = await response.Content.AsExpandoArrayAsync();
+                    
+                foreach (var orderId in eventData.OrdersDictionary.Keys)
+                {
+                    var order = eventData.OrdersDictionary[orderId];
+                    
+                    // For each order, fill in the product details. 
+                    var productDetail = ((IEnumerable<dynamic>)productList).First(product => product.productId == order.productId);
```

Comment:
Should this be `FirstOrDefault` and check for nulls later? Can it be that a product detail is missing?

Reply:
Since we expect the product to be there, I went for the First. If there's no product I did want it to throw.